src/appReview/slice.ts

```diff
@@ -0,0 +1,42 @@
+import { createSlice } from '@reduxjs/toolkit'
+import InAppReview from 'react-native-in-app-review'
+import { REHYDRATE, RehydrateAction } from 'redux-persist'
+import { getRehydratePayload } from 'src/redux/persist-helper'
+
+export interface State {
+  inAppRatingSupported: boolean
+  appRated: boolean
+  lastInteractionTimestamp: number | null
+}
+
+const initialState: State = {
+  // Android version >= 21 and iOS >= 10.3
+  // https://github.com/MinaSamir11/react-native-in-app-review#usage
+  inAppRatingSupported: !!InAppReview.isAvailable,
```

Comment:
Why do we want to store this in redux?

Reply:
I opted to keep it in redux so we can do a quick return in the other functions if the in app review isn't available.

Reply:
i think removing `initialized` and `inAppRatingSupported` from redux will simplify the logic, because then you'll also no longer need the `initAppReview` saga.

we'd only ever want to prompt someone to review the app if they've done something in the app before (i.e. dispatched some action, that you cannot dispatch until the wallet has been created already). am i missing a genuine scenario where the rating popup could be triggered during onboarding? 

i think the only variable that you need to persist in redux is the last app review interaction time + whether the user has already rated. on certain actions, the app rating saga can be spawned. this app rating saga will return early if 1) the OS level requirement isn't met, 2) if the user has rated before, 3) if the last interaction time doesn't meet the threshold. if none of these requirements are met, then the saga proceeds to launch the popup and record the last interaction time as the current time.

this app review saga should be triggered so rarely that you don't save much by checking the OS level and storing it in redux. whoever needs this information can do a real time check since it's a synchronous and fast check.  (i also think storing the `inAppRatingSupported` in redux opens a can of worms, since users can update their OS's. so you'd need to watch for that, which seems overcomplicated)

Reply:
+1 to what Kathy said.

We should only store the minimum necessary things in Redux.